backend/de.metas.async/src/main/java/de/metas/async/api/impl/AsyncBatchBL.java

```diff
@@ -486,20 +490,88 @@ public void markWorkpackageNotified(final I_C_Queue_WorkPackage_Notified workpac
 		return result.build();
 	}
 
+	@Override
+	public IAutoCloseable assignTempAsyncBatchIdToModel(@NonNull final Object model, @NonNull final AsyncBatchId asyncBatchId)
+	{
+		InterfaceWrapperHelper.setDynAttribute(model, DYN_ATTR_TEMPORARY_BATCH_ID, asyncBatchId);
+
+		return () -> InterfaceWrapperHelper.setDynAttribute(model, DYN_ATTR_TEMPORARY_BATCH_ID, null);
+	}
+
 	public I_C_Async_Batch getAsyncBatchById(@NonNull final AsyncBatchId asyncBatchId)
 	{
 		return asyncBatchDAO.retrieveAsyncBatchRecord(asyncBatchId);
 	}
 
 	@NonNull
+	@Override
 	public AsyncBatchId newAsyncBatch(@NonNull final String asyncBatchType)
 	{
 		final I_C_Async_Batch asyncBatch = trxManager.callInNewTrx(() -> newAsyncBatch()
 				.setContext(getCtx())
 				.setC_Async_Batch_Type(asyncBatchType)
 				.setName(asyncBatchType)
 				.build());
-
 		return AsyncBatchId.ofRepoId(asyncBatch.getC_Async_Batch_ID());
 	}
+
+	@Override
+	public Optional<String> getAsyncBatchTypeInternalName(@NonNull final I_C_Async_Batch asyncBatch)
+	{
+		return getAsyncBatchType(asyncBatch).map(AsyncBatchType::getInternalName);
+	}
+
+	@Override
+	public boolean isAsyncBatchTypeInternalName(@NonNull final I_C_Async_Batch asyncBatch, @NonNull final String expectedInternalName)
+	{
+		final String internalName = getAsyncBatchTypeInternalName(asyncBatch).orElse(null);
+		return internalName != null && internalName.equals(expectedInternalName);
```

Comment:
can be done later:
- note my comment about moving this method alltogether
- use Objects.equals. Its better readable